torch/_dynamo/variables/builder.py

```diff
@@ -920,3 +925,36 @@ def _clone_input(value):
             "torch.* op returned non-Tensor "
             + f"{typestr(example_value)} {proxy.node.op} {proxy.node.target}"
         )
+
+
+# Tracks the sources of all fake tensors we wrap in Dynamo.
+# Used by shape guard computation.
+@dataclasses.dataclass
+class FakeSource:
+    fake: Union[FakeTensor, SymInt]
+    sname: str
```

Comment:
nit: maybe put it in source.py?

Reply:
it's mostly here because GraphArgs is here, I don't have a big opinion (besides avoiding cycles)

Reply:
Hmm, actually there's some undesirable crosstalk here; this is not a subclass of Source. I'll give it another name.